var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -305,6 +305,13 @@ def setup_environment(self, spack_env, run_env):
             # MPI. It is also possible to set the variables in the modules.yaml
             # file if Intel MPI is the dominant, or only, MPI on a system.
             run_env.set('I_MPI_ROOT', join_path(self.prefix, 'impi'))
+            # Set mpi environment variables
+            spack_env.set('I_MPI_CC', spack_cc)
```

Comment:
You are probably right, I'll alter it.
What is the function signature? In the docs I see both:
```python
    def setup_dependent_environment(self, module, spec, dep_spec):
        """Dependencies of Qt find it using the QTDIR environment variable."""
        os.environ['QTDIR'] = self.prefix
```

The arguments to this function are:

    module: the module of the dependent package, where global properties can be assigned.
    spec: the spec of the dependency package (the one the function is called on).
    dep_spec: the spec of the dependent package (i.e. dep_spec depends on spec).

and
```python
def setup_dependent_environment(self, spack_env, run_env, extension_spec):
```

The difference in names confuses me.